src/main/resources/createJsonFilesLambda/createJsonFilesLambda.js

```diff
@@ -98,57 +95,142 @@ async function handleCreate(event, context) {
-    let totalObjectCount = 0;
-
-    // eslint-disable-next-line no-constant-condition
-    while (true) {
-        totalObjectCount += objectsPage.Contents.length;
-        if (objectsPage.Contents.length > 0) {
-            console.log(`Deleting ${objectsPage.Contents.length} objects from bucket '${event.bucket}'`);
-            const objectIds = objectsPage.Contents.map((object) => { return { Key: object.Key }; });
-            const param = { Bucket: event.bucket, Delete: { Objects: objectIds } };
-            const promise = doWithRetry(`Delete ${objectsPage.Contents.length} files`, () => s3.deleteObjects(param).promise());
-            promises.push(promise);
+    for (const chunk of chunks) {
+        if (chunk.length > 0) {
+            promises.push(invokeDeleteLambda(context, event.bucket, chunk));
             await delay(5);
         } else {
-            console.log('No more objects to delete, no need to call lambda');
+            console.log('Chunk is empty, nothing to delete');
         }
-        if (objectsPage.NextContinuationToken) {
-            objectsPage = await s3.listObjectsV2({
-                Bucket: event.bucket,
-                ContinuationToken: objectsPage.NextContinuationToken
-            }).promise();
-        } else {
-            break;
+    }
+    console.log(`Waiting for ${promises.length} lambdas to finish deleting ${allObjects.length} objects...`);
+    try {
+        await Promise.all(promises);
+        console.log(`Deleted ${allObjects.length} objects`);
+    } catch (error) {
+        context.fail('failed start delete-files lambda: ' + error);
+        throw error;
+    }
+}
+
+/**
+ * Invoke the lambda to delete the given S3 objects.
+ * @param {Context} context the lambda context
+ * @param {string} bucket the S3 bucket
+ * @param {string[]} objects the objects to delete
+ * @returns {Promise<unknown>} the invocation result
+ */
+async function invokeDeleteLambda(context, bucket, objects) {
+    /** @typedef {DeleteListEvent} */
+    const callParams = {
+        action: ACTION_DELETE_LIST,
+        bucket,
+        objects
+    };
+    console.log(`Calling lambda to delete ${objects.length} objects`);
+    const command = new InvokeCommand({
+        FunctionName: context.functionName,
+        Payload: new TextEncoder().encode(JSON.stringify(callParams)),
+        LogType: LogType.Tail
+    });
+    return lambdaClient.send(command);
+}
+
+/**
+ * Split the given array into chunks.
+ * @param {string[]} array the array to split
+ * @param {number} chunkSize the chunk size
+ * @returns {string[][]} the chunks
+ */
+function splitIntoChunks(array, chunkSize) {
+    const result = [];
+    for (let i = 0; i < array.length; i += chunkSize) {
+        const chunk = array.slice(i, i + chunkSize);
+        result.push(chunk);
+    }
+    return result;
+}
+
+/**
+ * Fetch all S3 objects from a bucket.
+ * @param {string} bucket the S3 bucket
+ * @returns {Promise<string[]>} the keys of all objects
+ */
```

Comment:
Could we rename the function to `listS3BucketContents()` or `listContents()` if you prefer.
And add some content that this is required (or "used") for deleting the objects, i.e. empty the bucket?

Reply:
Is it required to use the term "object" here? Aren't the contents of a bucket rather "files"? Or could it be something else as well?